core/src/main/java/org/infinispan/configuration/cache/MemoryConfigurationBuilder.java

```diff
@@ -21,34 +24,70 @@
 public class MemoryConfigurationBuilder extends AbstractConfigurationChildBuilder implements Builder<MemoryConfiguration>, ConfigurationBuilderInfo {
    private MemoryStorageConfigurationBuilder memoryStorageConfigurationBuilder;
    private final List<ConfigurationBuilderInfo> elements;
+   private final AttributeSet attributes;
 
    MemoryConfigurationBuilder(ConfigurationBuilder builder) {
       super(builder);
       this.memoryStorageConfigurationBuilder = new MemoryStorageConfigurationBuilder(builder);
       this.elements = Collections.singletonList(memoryStorageConfigurationBuilder);
+      this.attributes = MemoryConfiguration.attributeDefinitionSet();
    }
 
    @Override
    public Collection<ConfigurationBuilderInfo> getChildrenInfo() {
       return elements;
    }
 
+   @Override
+   public AttributeSet attributes() {
+      return attributes;
+   }
+
    /**
     * Set the {@link StorageType} to determine how the data is stored in the data container.
     * @param storageType the storage type of the underlying data
     * @return this configuration builder
     */
    public MemoryConfigurationBuilder storageType(StorageType storageType) {
-      memoryStorageConfigurationBuilder.storageType(storageType);
+      StorageType config = storageType == StorageType.OBJECT ? StorageType.HEAP : storageType;
+      if (config == StorageType.BINARY) {
+         encoding().mediaType(MediaType.APPLICATION_PROTOSTREAM_TYPE);
```

Comment:
Here I am dealing with a deprecated BINARY setting. In ISPN 10 and master, users would set this and the end result is the default marshaller would be used to store the entries (protostream). I am just short-circuiting the whole thing here since I have no idea (at config time) what is the default marshaller, since it is calculated dynamically.

Reply:
A more correct implementation maybe would be to defer this decision to when the DataConversion is created

Reply:
Yeah I am not sure what the best approach here is. I just am a bit worried that setting it to Protostream when it isn't can cause issues for content negotiation down the line. I know we don't want to break backwards compatibility in this version, but maybe we have to to be safe and instead throw an exception if we can't determine the media type?

Reply:
I will remove this MediaType from here, deferring this logic to runtime rather than config time, so it should pick either the configured one, or the one from the user marshaller